### PR TITLE
Add TPC-H 100GB on TiDB 5.0 with MPP benchmark

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added or deleted? (Required)

Add TiDB v5.0 MPP benchmark, which is compared with Greenplum and Apache Spark.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/5956
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: benchmark/v5.0-performance-benchmarking-with-tpch.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+---
+title: TiDB TPC-H 100GB Performance Test Report -- v5.0 MPP mode vs. Greenplum 6.15.0 and Apache Spark 3.1.1
+summary: Compare the TPC-H 100GB performance of TiDB 5.0 MPP mode, Greenplum 6.15.0, and Apache Spark 3.1.1.
+---
+
+# TiDB TPC-H 100GB Performance Test Report -- TiDB v5.0 MPP mode vs. Greenplum 6.15.0 and Apache Spark 3.1.1
+
+## Test overview
+
+This test aims to compare the TPC-H 100 performance of TiDB v5.0 in the MPP mode with that of two mainstream analytics engines, Greenplum and Apache Spark, in their latest version. The test result shows that the performance of TiDB v5.0 in the MPP mode is two to three times better than other two solutions.
```
Comment (qiancai) [inline benchmark/v5.0-performance-benchmarking-with-tpch.md:10]:
```suggestion
This test aims at comparing the TPC-H 100 performance of TiDB v5.0 in the MPP mode with that of two mainstream analytics engines, Greenplum and Apache Spark, in their latest versions. The test result shows that the performance of TiDB v5.0 in the MPP mode is two to three times better than the other two solutions.
```

File: benchmark/v5.0-performance-benchmarking-with-tpch.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+--total-executor-cores 120
+--executor-cores 5
+--executor-memory 15G
+```
+
+## Test result
+
+> **Note:**
+>
+> The following test results are the average of three tests. All numbers are in seconds.
```
Comment (qiancai) [inline benchmark/v5.0-performance-benchmarking-with-tpch.md:89]:
```suggestion
> The following test results are the average data of three tests. All numbers are in seconds.
```

File: benchmark/v5.0-performance-benchmarking-with-tpch.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+| 22      |    1.23    |   7.67476   |       4.59    |
+
+![TPC-H](/media/tidb-v5-tpch-100-vs-gp-spark.png)
+
+In the performance diagram above:
+
+- Blue lines represent TiDB v5.0;
+- Red lines represent Greenplum 6.15.0;
+- Yellow lines represent Apache Spark 3.1.1.
+- The y-axis represents the execution time of the query. The less the time, the better the performance.
```
Comment (qiancai) [inline benchmark/v5.0-performance-benchmarking-with-tpch.md:123]:
```suggestion
- The y-axis represents the execution time of the query. The less the time, the better the performance.
```
```suggestion
- The y-axis represents the execution time of the query. The less the time is, the better the performance is.
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

File: benchmark/v5.0-performance-benchmarking-with-tpch.md (hunk 1 above)
Comment (CharLotteiu) [inline benchmark/v5.0-performance-benchmarking-with-tpch.md]:
```suggestion
This test aims to compare the TPC-H 100GB performance of TiDB v5.0 in the MPP mode with that of Greenplum and Apache Spark, two mainstream analytics engines, in their latest version. The test result shows that the performance of TiDB v5.0 in the MPP mode is two to three times better than that of the other two solutions.
```

File: TOC.md [hunk 4]
```diff
@@ -9,7 +9,9 @@
   + [TiDB Introduction](/overview.md)
   + [What's New in TiDB 5.0](/releases/release-5.0.0-rc.md)
   + [Basic Features](/basic-features.md)
+  + [Experimental Features](/experimental-features.md)
```
Comment (yikeke) [inline TOC.md:12]: Forgot to add this in another PR. Please take it together.

File: benchmark/v5.0-performance-benchmarking-with-tpch.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
+---
+title: TiDB TPC-H 100GB Performance Test Report -- v5.0 MPP mode vs. Greenplum 6.15.0 and Apache Spark 3.1.1
+summary: Compare the TPC-H 100GB performance of TiDB 5.0 MPP mode, Greenplum 6.15.0, and Apache Spark 3.1.1.
+---
+
+# TiDB TPC-H 100GB Performance Test Report -- TiDB v5.0 MPP mode vs. Greenplum 6.15.0 and Apache Spark 3.1.1
+
+## Test overview
+
+This test aims at comparing the TPC-H 100 performance of TiDB v5.0 in the MPP mode with that of Greenplum and Apache Spark, two mainstream analytics engines, in their latest versions. The test result shows that the performance of TiDB v5.0 in the MPP mode is two to three times better than that of the other two solutions.
```
Comment (CharLotteiu) [inline benchmark/v5.0-performance-benchmarking-with-tpch.md:10]:
```suggestion
This test aims at comparing the TPC-H 100GB performance of TiDB v5.0 in the MPP mode with that of Greenplum and Apache Spark, two mainstream analytics engines, in their latest versions. The test result shows that the performance of TiDB v5.0 in the MPP mode is two to three times better than that of the other two solutions.
```

File: benchmark/v5.0-performance-benchmarking-with-tpch.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
+---
+title: TiDB TPC-H 100GB Performance Test Report -- v5.0 MPP mode vs. Greenplum 6.15.0 and Apache Spark 3.1.1
+summary: Compare the TPC-H 100GB performance of TiDB 5.0 MPP mode, Greenplum 6.15.0, and Apache Spark 3.1.1.
+---
+
+# TiDB TPC-H 100GB Performance Test Report -- TiDB v5.0 MPP mode vs. Greenplum 6.15.0 and Apache Spark 3.1.1
+
+## Test overview
+
+This test aims at comparing the TPC-H 100GB performance of TiDB v5.0 in the MPP mode with that of Greenplum and Apache Spark, two mainstream analytics engines, in their latest versions. The test result shows that the performance of TiDB v5.0 in the MPP mode is two to three times better than that of the other two solutions.
```
Comment (ilovesoup) [inline benchmark/v5.0-performance-benchmarking-with-tpch.md:10]: "two to three times better" is not a proper description. Should be 
"2 to 3 times faster than that of the other two solutions under TPC-H workload"

File: benchmark/v5.0-performance-benchmarking-with-tpch.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+In v5.0, TiDB introduces the MPP mode for [TiFlash](/tiflash/tiflash-overview.md), which significantly enhances TiDB's Hybrid Transactional and Analytical Processing (HTAP) capabilities. Test objects in this report are as follows:
+
++ TiDB v5.0 column-based storage in the MPP mode
```
Comment (ilovesoup) [inline benchmark/v5.0-performance-benchmarking-with-tpch.md:14]:
```suggestion
+ TiDB v5.0 columnar storage in the MPP mode
```

File: benchmark/v5.0-performance-benchmarking-with-tpch.md (hunk 6 above)
Comment (ilovesoup) [inline benchmark/v5.0-performance-benchmarking-with-tpch.md:10]:
```suggestion
This test aims at comparing the TPC-H 100GB performance of TiDB v5.0 in the MPP mode with that of Greenplum and Apache Spark, two mainstream analytics engines, in their latest versions. The test result shows that the performance of TiDB v5.0 in the MPP mode is two to three times faster than that of the other two solutions under TPC-H workload.
```

File: benchmark/v5.0-performance-benchmarking-with-tpch.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+Set session variables with the following SQL statements:
+
+```sql
+set @@tidb_isolation_read_engines='tiflash';
+set @@tidb_allow_mpp=1;
+set @@tidb_opt_broadcast_join=0;
+set @@tidb_mem_quota_query = 10 << 30;
+```
+
+All TPC-H test tables replicated to TiDB are stored in column-based TiFlash, with no additional partitions or indexes.
```
Comment (ilovesoup) [inline benchmark/v5.0-performance-benchmarking-with-tpch.md:57]:
```suggestion
All TPC-H test tables are replicated to TiFlash in columnar format, with no additional partitions or indexes.
```

File: benchmark/v5.0-performance-benchmarking-with-tpch.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+```
+log_statement = all
+gp_autostats_mode = none
+statement_mem = 2048MB
+gp_vmem_protect_limit = 16384
+```
+
+#### Apache Spark
+
+The test of Apache Spark uses Apache Parquet as the storage format and stores the data on HDFS. The HDFS system consists of three nodes. Each node has two assigned NVMe SSD disks as the data disks. The Spark cluster is started standalone, using NVMe SSD disks as the local directory of `spark.local.dir` to speed up the shuffle spill, with no additional partitions or indexes.
```
Comment (ilovesoup) [inline benchmark/v5.0-performance-benchmarking-with-tpch.md:74]:
```suggestion
The test of Apache Spark uses Apache Parquet as the storage format and stores the data on HDFS. The HDFS system consists of three nodes. Each node has two assigned NVMe SSD disks as the data disks. The Spark cluster is deployed in standalone mode, using NVMe SSD disks as the local directory of `spark.local.dir` to speed up the shuffle spill, with no additional partitions or indexes.
```

**Review by zanmato1984 — APPROVED** [review]
LGTM

**Pull request merged**